### PR TITLE
feat(dynamo): plug things with localstack

### PR DESCRIPTION
**Pull request by JacopoDaeli**

### CODE REVIEW

File: bin/dynamo-tables.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+      },
+      {
+        AttributeName: 'env',
+        AttributeType: 'S'
+      }
+    ],
+    ProvisionedThroughput: {
+      ReadCapacityUnits: 5,
+      WriteCapacityUnits: 5
+    }
```
Comment (decompil3d) [inline bin/dynamo-tables.js:28]: This feels like it should be configurable, as it'll vary based on the specific needs of the organization using Warehouse.

File: bin/dynamo-tools.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+const { tables, tableNames } = require('./dynamo-tables');
+
+class DynamoTools {
+  constructor({ clients, regions }) {
+    this._clients = clients;
+    this._regions = regions;
+  }
+
+  async getTableStatus(region, tableName) {
```
Comment (decompil3d) [inline bin/dynamo-tools.js:13]: JSDoc throughout

File: bin/init-localstack.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+const { DynamoDB } = require('aws-sdk');
+const DynamoTools = require('./dynamo-tools');
+
+const regions = ['us-west-2'];
+const clients = regions.reduce((acc, region) => {
+  acc[region] = new DynamoDB({
+    accessKeyId: 'fakeKeyId',
+    secretAccessKey: 'fakeSecretAccessKey',
+    endpoint: 'http://localhost:4569',
```
Comment (decompil3d) [inline bin/init-localstack.js:11]: Latest version of `localstack` runs all services on port 4566 only.

```suggestion
    endpoint: 'http://localhost:4566',
```

File: lib/config.js [hunk 4]
```diff
@@ -3,9 +3,20 @@ const fp = require('fastify-plugin');
...
-    });
+    const config = {
+      port: process.env.PORT || 8088,
+      dynamo: {}
+    };
+
+    const localstack = process.env.LOCALSTACK || 0;
+    if (localstack) {
+      config.dynamo = {
+        endpoint: process.env.LOCALSTACK_STS_URL || 'http://localhost:4569',
```
Comment (decompil3d) [inline lib/config.js:14]:
```suggestion
        endpoint: process.env.LOCALSTACK_STS_URL || 'http://localhost:4566',
```

File: package.json [hunk 5]
```diff
@@ -29,11 +29,12 @@
...
     "lint": "npm run prettier && npm run eslint",
     "eslint": "eslint-godaddy --max-warnings 0 --ignore-pattern /coverage/ ./",
-    "prettier": "prettier --write lib",
+    "prettier": "prettier --write lib bin",
     "release": "standard-version -t ''",
     "test": "tap test/**/*.test.js",
     "start": "node lib/app.js",
-    "dev": "nodemon lib/app.js"
+    "dev": "LOCALSTACK=1 AWS_ACCESS_KEY_ID=foobar AWS_SECRET_ACCESS_KEY=foobar nodemon lib/app.js",
+    "localstack": "docker run -it -p 4569:4569 -p 9999:8080 -e SERVICES=dynamodb -e DEBUG=1 --rm localstack/localstack:0.10.5"
```
Comment (decompil3d) [inline package.json:37]:
```suggestion
    "localstack": "docker run -it -p 4566:4566 -e SERVICES=dynamodb -e DEBUG=1 --rm localstack/localstack"
```

**Review by decompil3d — CHANGES_REQUESTED** [review]

File: bin/init-localstack.js (hunk 3 above)
Comment (JacopoDaeli) [reply bin/init-localstack.js:11]: Not using the latest version of localstack.

File: package.json (hunk 5 above)
Comment (JacopoDaeli) [reply package.json:37]: We should use a specific version of localstack. Never the latest. I had very bad experience using the latest one.

Comment (JacopoDaeli) [reply package.json:37]: Localstack is well know for breaking things on new releases. It happened to me twice in the past. version `0.10.5` is pretty stable and it's what we use in kubernetes-external-secrets.

Comment (decompil3d) [reply package.json:37]: I think we should take the latest one as of now, pin to that minor version (e.g. 0.11) and update as needed later.

File: bin/dynamo-tables.js (hunk 1 above)
Comment (JacopoDaeli) [reply bin/dynamo-tables.js:28]: At the moment, this is just for local stuff. It really doesn't matter.

File: bin/dynamo-tools.js (hunk 2 above)
Comment (JacopoDaeli) [reply bin/dynamo-tools.js:13]: This is just for tooling / local development stuff so not a very high priority. But sure will add.

File: package.json (hunk 5 above)
Comment (JacopoDaeli) [reply package.json]: Done.

File: bin/dynamo-tables.js (hunk 1 above)
Comment (JacopoDaeli) [reply bin/dynamo-tables.js:28]: I mean these stuff is to create tables locally on local stuff for now.

File: bin/dynamo-tools.js (hunk 2 above)
Comment (JacopoDaeli) [reply bin/dynamo-tools.js:13]: Right now this is used only to create the tables in localstack.

**Review by decompil3d — APPROVED** [review]

**Pull request merged**